Repositories/ProjectRepository.cs

```diff
@@ -175,9 +179,24 @@ public override async Task<Project> FindAsync(int id)
             List<Project> projectResults = await queryableProjects.ToListAsync();
 
             //Redact the user after fetching the collection from the project (no separate query needs to be executed)
-            projectResults.ForEach( project => project.User = RedactUser( project.User ) );
+            projectResults.ForEach(project => project.User = RedactUser(project.User));
 
-            return projectResults;
+            // This is needed because otherwise likes, collaborators and categories are missing!
+            foreach(Project project in queryableProjects)
+            {
+                project.Collaborators = await GetDbSet<Collaborator>()
+                                              .Where(p => p.ProjectId == project.Id)
+                                              .ToListAsync();
+                project.User = RedactUser(project.User);
+                project.Likes = await GetDbSet<ProjectLike>()
+                                      .Where(p => p.LikedProject.Id == project.Id)
+                                      .ToListAsync();
+                project.Categories = await GetDbSet<ProjectCategory>()
+                                      .Include(p => p.Category)
+                                      .Where(p => p.Project.Id == project.Id)
+                                      .ToListAsync();
+            }
+            return await queryableProjects.ToListAsync();
```

Comment:
Not that it matters to much at this time but this could become really bad in the future performance wise then it might be better to create custom queries if the need arises

Reply:
Can you make a suggestion on how you would improve this?

Reply:
I would make a function that would get a project with those specific relations that executes custom writen and optimized sql.  Since the documentation says the following:  "Raw SQL queries are also used if using a LINQ query is resulting in an inefficient SQL query." See https://docs.microsoft.com/en-us/ef/core/querying/raw-sql

Reply:
Seems like a good idea. Maybe I'll pick it up myself in a follow-up issue, since it's a bit out of scope here. @rubenb994 what do you think? Should I make a new issue?

Reply:
@JVerbruggen For me, you could either resolve it in this issue because this method is used in this issue scope as well, or make a new issue if you would like to tackle all methods in the repository layer to make sure the performance is ok. (I assume you meant to tag me by the way :smile: ?)